app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragment.kt

```diff
@@ -85,32 +113,20 @@ class CreditCardsSettingFragment : PreferenceFragmentCompat() {
                 .getString(R.string.preferences_credit_cards_sync_cards_across_devices),
             loggedInTitle = requireContext()
                 .getString(R.string.preferences_credit_cards_sync_cards),
-            onSignInToSyncClicked = {
-                val directions =
-                    CreditCardsSettingFragmentDirections.actionCreditCardsSettingFragmentToTurnOnSyncFragment()
-                findNavController().navigateBlockingForAsyncNavGraph(directions)
-            },
-            onReconnectClicked = {
-                val directions =
-                    CreditCardsSettingFragmentDirections.actionGlobalAccountProblemFragment()
-                findNavController().navigateBlockingForAsyncNavGraph(directions)
-            }
+            onSignInToSyncClicked = { navigateToSyncFragment() },
+            onReconnectClicked = { navigateToAccountProblemFragment() }
         )
+
+        togglePrefsEnabledWhileAuthenticating(creditCardPreferences, true)
     }
 
     override fun onPreferenceTreeClick(preference: Preference): Boolean {
         when (preference.key) {
             getPreferenceKey(R.string.pref_key_credit_cards_add_credit_card) -> {
-                val directions =
-                    CreditCardsSettingFragmentDirections
-                        .actionCreditCardsSettingFragmentToCreditCardEditorFragment()
-                findNavController().navigateBlockingForAsyncNavGraph(directions)
+                navigateToCreditCardEditorFragment()
```

Comment:
Same as above. I think inlining is fine unless the navigateX function is reusable and needs to be in a callback.